src/Features/Core/Portable/PullMemberUp/AbstractPullMemberUpRefactoringProvider.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PullMemberUp.QuickAction;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp
+{
+    internal abstract class AbstractPullMemberUpRefactoringProvider : CodeRefactoringProvider
+    {
+        private CancellationToken _cancellationToken;
+
+        protected abstract bool IsSelectionValid(TextSpan span, SyntaxNode userSelectedSyntax);
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            // Currently support to pull field, method, event, property and indexer up,
+            // constructor, operator and finalizer are excluded.
+            _cancellationToken = context.CancellationToken;
```

Comment:
this is not safe.  yoy might be called by multiple threads simulataneously.  You will be slamming over another thread's cancellation token.  If you want to do this sort of thing, you need to instantiate a new object here that holds onto the cancellation token and move your methods into it.  That way each thread gets a fresh object they can work with.

Reply:
Yes, will remove it in next commit